const-directives.md

```diff
@@ -2,8 +2,210 @@
+| RG8UI   | RG16UI   | RG32UI   |
+| RGB8UI  | RGB16UI  | RGB32UI  |
+| RGBA8UI | RGBA16UI | RGBA32UI |
+
+
+This directive only needs to be defined once in the shader pack, can can be defined in (mostly) any shader file. Because the format names are not defined as part of glsl, the directive must either be put in a block comment, or the format name must be otherwise manually defined to any value to avoid compilation errors. Do not put the directive in a single line comment, or have any other text in the line of the directive, as this may cause it to not be detected properly.
+
+### Declaration
+
+Scope: Entire Pack
+
+```glsl
+/*
+const int <bufferName>Format = <format>;
+
+Ex:
+const int colortex4Format = RGB9_E5;
+*/
+```
+
+### Valid Declaration Locations
+
+* ❌ Vertex Shader (*.vsh)
+* ❌ Geometry Shader (*.gsh)
+* ✔️ Fragment Shader (*.fsh)
+
+### Implementation Support
+
+* ❓ ShadersMod
+* ✔️ OptiFine
+* ✔️ Iris
+
+## Texture Clear
+
+This Clear directive allows a shader pack to disable clearing for a colortex or shadowcolor color attachment. This means that the values written to the texture will be retained through the next frame instead of being cleared after every frame.
+
+The ClearColor directive is only used when clearing is enabled, and it defined the values stored in the buffer during the clear operation. By default, all colortex buffers are cleared to 0s (black), except colortex0, which is cleared to the current fog color, and colortex1, which is cleared to white. All shadowcolor buffers are cleared with white by default. This directive allows changing this clear color for each attachment.
+
+Both of these directives only need to be defined once in the shader pack, can can be defined in (mostly) any shader file.
+
+### Declaration
+
+Scope: Entire Pack
+
+```glsl
+const bool <bufferName>Clear = false;
+const vec4 <bufferName>ClearColor = <value>;
+```
+
+### Value Range
+
+* Clear
+	* Possible Values: true/false
+	* Default Value: false
+	* Out-of-range values behavior: default value (false)
+* ClearColor
+	* Possible Values: true/false
+	* Default Value: vec4(0.0, 0.0, 0.0, 0.0)
+	* Out-of-range values behavior: default value
+
+### Valid Declaration Locations
+
+* ❌ Vertex Shader (*.vsh)
+* ❌ Geometry Shader (*.gsh)
+* ✔️ Fragment Shader (*.fsh)
+
+### Implementation Support
+
+* ❓ ShadersMod
+* ✔️ OptiFine
+* ✔️ Iris
+
+## Texture Mipmaps
+
+This directive allows shaders to generate a mipmap chain for a colortex attachment. The mipmap chain will be generated before any program where the directive is included, if there are multiple programs it will be re-generated before each program containing the directive. This directive is only valid for composite, deferred, and final programs.
+
+### Declaration
+
+Scope: Per-program
+
+```glsl
+const bool <bufferName>MipmapEnabled = true;
+```
+
+### Value Range
+
+* Possible Values: true/false
+* Default Value: false
+* Out-of-range values behavior: default value (false)
```

Comment:
Curious if this is the actual behavior or if the directive would merely be ignored / flag an error or warning?

Reply:
Yep, in that case I'd suggest something like "Ignored (warning logged)" since that means that this directive won't compete for precedence with a valid one, unlike Default which would. Same for anywhere else where that applies.